components/cloud/azure/src/token_credentials_with_cache/secret_credentials.rs

```diff
@@ -0,0 +1,65 @@
+// Copyright 2023 TiKV Project Authors. Licensed under Apache-2.0.
+
+use std::str;
+
+use azure_core::{
+    auth::{TokenCredential, TokenResponse},
+    new_http_client,
+};
+use azure_identity::{ClientSecretCredential, TokenCredentialOptions};
+
+use super::utils::{TokenCache, TokenCredentialWithCache};
+
+/// Enables authentication to Azure Active Directory using a client secret that
+/// was generated for an App Registration. It will automatically cache the
+/// latest token from Azure Active Directory.
+pub struct ClientSecretCredentialWithCache {
+    credential: ClientSecretCredential,
+
+    /// Cached token
+    cached_token: TokenCache,
```

Comment:
Why we need the cache?

Reply:
To reduce the cost of requesting a new TOKEN from Azure, this operation costs a lot (seconds level latency). 
And a new token returned from Azure keeps 1 hour ttl as default. That is, during the ttl, we could cache this token to reduce the cost on requesting new token.